### DISCUSSION TITLE
Workflows Expressions dialog > php functions

### DISCUSSION THREAD

**Discussion by s2b-git** [YetiForceCompany/YetiForceCRM#14996] [How to] [answered]
Workflows Expressions dialog box contains only "concat" funtion working on text. Is there any other basic php functions what will work in custom expression box?

**Comment by mariuszkrzaczkowski** [comment, marked as answer]
https://github.com/YetiForceCompany/YetiForceCRM/blob/e0d22cea149f977acb275b603d97b264f7753d3e/modules/com_vtiger_workflow/expression_engine/VTExpressionsManager.php#L79L84

**Reply by s2b-git** [reply]
Thanks!

**Comment by JohnMaverick** [comment]
Tried include str_replace but it dont work. How to remove spaces from text field?

**Reply by mariuszkrzaczkowski** [reply]
show sample value

**Reply by JohnMaverick** [reply]
preg_replace('/\s+/', '','textfield1'); end

**Reply by JohnMaverick** [reply]
concat function working well, preg_replace added to VTExpressionManager. Tried with trim etc... doesnt work too

**Reply by vovpff** [reply]
@s2b-git @JohnMaverick At first you need to add new functions [here]( https://github.com/YetiForceCompany/YetiForceCRM/blob/e0d22cea149f977acb275b603d97b264f7753d3e/modules/com_vtiger_workflow/expression_engine/VTExpressionEvaluater.php#L80)
Next you need to add matching between your new php function with YF system [here](https://github.com/YetiForceCompany/YetiForceCRM/blob/e0d22cea149f977acb275b603d97b264f7753d3e/modules/com_vtiger_workflow/expression_engine/VTExpressionEvaluater.php#L217) and [here](https://github.com/YetiForceCompany/YetiForceCRM/blob/e0d22cea149f977acb275b603d97b264f7753d3e/modules/com_vtiger_workflow/expression_engine/VTExpressionsManager.php#L79L84)

And keep in mind what after system update your code can be overwritten by update package files and this cause a lot of errors.

**Reply by mariuszkrzaczkowski** [reply]
I recommend adding your changes to the system using Pull Requests, it will let you avoid problems when updating the system.

**Comment by garryatbrighton** [comment]
would be cool to allow full php calls in expressions.
